splitgpg2/__init__.py

```diff
@@ -0,0 +1,804 @@
+        args = self.verify_keygrip_arguments(1, 1, untrusted_args)
+        await self.agent_protocol.send_command(b'SETKEY', args)
+
+    async def command_SETKEYDESC(self, untrusted_args: Optional[bytes]):
+        # XXX: is there a better way than showing the message
+        #      from the untrusted domain
+        args = self.sanitize_key_desc(untrusted_args)
+        await self.agent_protocol.send_command(b'SETKEYDESC', args)
+
+    async def command_PKDECRYPT(self, untrusted_args: Optional[bytes]):
+        if untrusted_args:
+            raise Filtered
+        self.request_timer('PKDECRYPT')
+        await self.agent_protocol.send_command(b'PKDECRYPT', None)
+
+    async def command_SETHASH(self, untrusted_args: Optional[bytes]):
+        untrusted_alg, untrusted_hash = untrusted_args.split(b' ', 1)
+        try:
+            alg = int(untrusted_alg)
+            alg_param = self.hash_algos[alg]
+        except (KeyError, ValueError):
+            raise Filtered
+
+        if not untrusted_hash:
+            raise Filtered
+
+        hash_regex = re.compile(rb'[0-9A-F]{%d}' % alg_param.len)
+        if hash_regex.fullmatch(untrusted_hash) is None:
+            raise Filtered
+        hash_value = untrusted_hash
+
+        await self.agent_protocol.send_command(
+            b'SETHASH', b'%d %s' % (alg, hash_value))
+
+    async def command_PKSIGN(self, untrusted_args: Optional[bytes]):
+        if untrusted_args:
+            if untrusted_args.startswith(b'-- '):
+                untrusted_args = untrusted_args[3:]
+            if self.cache_nonce_regex.fullmatch(untrusted_args) is None:
+                raise Filtered
+            args = b'-- ' + untrusted_args
+        else:
+            args = None
+
+        self.request_timer('PKSIGN')
+
+        await self.agent_protocol.send_command(b'PKSIGN', args)
+
+    async def command_GETINFO(self, untrusted_args: Optional[bytes]):
+        if untrusted_args != b'version':
+            raise Filtered
+        args = untrusted_args
+
+        await self.agent_protocol.send_command(b'GETINFO', args)
+
+    async def command_BYE(self, untrusted_args: Optional[bytes]):
+        if untrusted_args:
+            raise Filtered
+        await self.agent_protocol.send_command(b'BYE', None)
+
+
+class AgentProtocol(asyncio.Protocol):
+    """
+    Protocol class for interacting with a local (real) gpg agent.
+    """
+    # type hints, enable when python >= 3.6 will be everywhere...
+    # agent_responded: Optional[Future]
+    # expected_inquires: Dict[bytes, Callable]
+    # untrusted_agent_response_buffer: bytearray
+    # server: ServerProtocol
+
+    def __init__(self, server: ServerProtocol) -> None:
+        super(AgentProtocol, self).__init__()
+        self.server = server
+        self.agent_transport = None
+        # We gennerally consider the agent as trusted. But since the client can
+        # determine part of the response we handle this here as untrusted.
```

Comment:
I would move this comment to `handle_agent_response`

Reply:
Ok.